### PR TITLE
Add conditions for sidecar auditLog.destination in deployment.yaml

### PR DESCRIPTION
**Pull request by dkeightley**
Create the sidecar container only when `auditLog.destination: sidecar` is passed as a value without a `auditLog.hostPath`.

This achieves the outcome expected as per our documentation, where destination `hostPath` and `sidecar` values are mutually exclusive.

A user can still make these mutually inclusive by passing a `hostPath`, for eg:

```yaml
auditLog:
  destination: sidecar
  hostPath: /var/log/testing/
  level: 2
```

Tested chart with Rancher v2.3.5:
- `auditLog.destination: hostPath` no longer creates a sidecar
- `auditLog.destination: sidecar` and `auditLog.hostPath` creates a sidecar and logs to a specified directory

gz#16192
gz#17090

### CODE REVIEW

File: chart/templates/deployment.yaml [hunk 1]
```diff
@@ -167,7 +169,7 @@ spec:
           secretName: tls-ca
 {{- end }}
 {{- if gt (int .Values.auditLog.level) 0 }}
-  {{- if eq .Values.auditLog.destination "hostPath" }}
+  {{- if .Values.auditLog.hostPath }}
```
Comment (loganhz) [inline chart/templates/deployment.yaml:172]: Please just keep `if eq .Values.auditLog.destination "hostPath" `. We don't need to change this line.

there is a default value of `.Values.auditLog.hostPath`. If you do so, no matter if user is using `hostPath` mode, Rancher will write audit logs to the default hostPath on the node.

Comment (dkeightley) [reply chart/templates/deployment.yaml]: Done

**Review by loganhz — DISMISSED** [review]

**Review by rawmind0 — APPROVED** [review]
LGTM

**Comment by rawmind0** [comment]
CI is failing on master branch due to PR https://github.com/rancher/rancher/pull/35157 was not back ported

`Package 'docker-20.10.6_ce-6.49.3' not found.`

**Comment by rawmind0** [comment]
As master branch is not active anymore, this PR is superseded by: 
* 2.6: https://github.com/rancher/rancher/pull/35805 
* 2.5: https://github.com/rancher/rancher/pull/35806

**Pull request closed without merging**